### PR TITLE
Restructure build system

### PR DESCRIPTION
**Pull request by thp**
This pull request is split off from #10 and contains only the build system changes that will later on enable the Continous Integration builds on Travis CI.

### CODE REVIEW

**Comment by karcherm** [comment]
This pull request does not only contain build system changes, but also changes the include directives in qhimdwindetection.cpp. It seems these changes are appropriate if you use the original Microsoft header files. When that file was written, the subdirectory "ddk" was indeed needed if you (cross-)compile with mingw using the header files mingw provides.
Did that change in the mean time?

**Comment by thp** [comment]
I needed to do that change so that it compiles in Ubuntu (as on Travis CI) with Mingw-W64 (which is a fork of mingw32 with support for both 32- and 64-bit). I didn't test that change with Windows compilers, though.

Here's the build logs for building with 32- and 64-bit versions of Mingw-W64 with that code change:

https://travis-ci.org/thp/linux-minidisc/jobs/163859833
https://travis-ci.org/thp/linux-minidisc/jobs/163859831

**Comment by thp** [comment]
If you really want, I can split out the changes to `qhimdtransfer/qhimdwindetection.cpp` to a separate pull request, just let me know if there's anything left to do here or if this can be merged.

**Comment by thp** [comment]
_bump_ - what's the status here, should I separate out the .cpp changes into a separate pull request or can we merge this as-is?

**Comment by glaubitz** [comment]
Yes, those should go into a different commit, please.

**Comment by thp** [comment]
Spilt off into https://github.com/glaubitz/linux-minidisc/pull/22 and rebased this one on top of it.

**Comment by thp** [comment]
Merge after #22.

File: qhimdtransfer/qhimdwindetection.cpp [hunk 1]
```diff
@@ -5,10 +5,10 @@
 
 #include <windows.h>
 #include <dbt.h>
-#include <setupapi.h>
 #include <initguid.h>       // needed to handle GUIDs
-#include <ddk/ntddstor.h>   // needed for handling storage devices
-#include <ddk/cfgmgr32.h>   // needed for CM_Get_Child function
+
+#include <ntddstor.h>   // needed for handling storage devices
+#include <cfgmgr32.h>   // needed for CM_Get_Child function
```
Comment (glaubitz) [inline qhimdtransfer/qhimdwindetection.cpp:11]: Why is this still in here? Wasn't this supposed to be in #22?

File: build/libgcrypt.pri [hunk 2]
```diff
@@ -0,0 +1,6 @@
+!without_gcrypt {
+  LIBS += -lgcrypt
+  DEFINES += CONFIG_WITH_GCRYPT
+}
+else: !build_pass: message(You disabled mcrypt: No PCM and ATRAC transfer will be supported)
```
Comment (glaubitz) [inline build/libgcrypt.pri:5]: The message should mention "gcrypt", not "mcrypt". mcrypt was dropped in c5a16774418f25ff30089589c8249d3708c8335c.

**Review by glaubitz — CHANGES_REQUESTED** [review]

File: qhimdtransfer/qhimdwindetection.cpp (hunk 1 above)
Comment (glaubitz) [reply qhimdtransfer/qhimdwindetection.cpp:11]: Ok, this change doesn't show up in the other view. Strange.

**Pull request merged**